tmt/steps/provision/__init__.py

```diff
@@ -1103,6 +1108,42 @@ def requires(cls) -> list['tmt.base.Dependency']:
         """ All requirements of the guest implementation """
         return []
 
+    def acquire_log(self, log_name: str) -> Optional[str]:
+        """fetch and return content of a requested log"""
+        raise NotImplementedError
+
+    def store_log(
+            self,
+            log_path: Path,
+            log_content: str,
+            log_name: Optional[str] = None) -> None:
+        """save log content and return the path"""
+        if log_name:
+            # if log_path contains log file name
+            if log_path.is_dir():
+                log_path.write_text(log_content)
+            else:
+                (log_path / log_name).write_text(log_content)
+        else:
+            log_path.write_text(log_content)
+
+    def handle_guest_logs(self,
+                          log_path: Optional[Path] = None,
+                          logs: Optional[list[str]] = None) -> None:
+        """get log content and save it to the workdir/provision/,
+        list the log dir in guests.yaml"""
+        logs = logs or []
```

Comment:
It might be better to initialize with `self.logs`, right? Once it returns an empty list...